packages/embark-web3/src/index.js

```diff
@@ -17,91 +15,55 @@ class EmbarkWeb3 {
     this.embark = embark;
     this.logger = embark.logger;
     this.events = embark.events;
-    this.fs = embark.fs;
     this.config = embark.config;
 
-    this.events.request("runcode:whitelist", 'web3', () => { });
+    this.setupWeb3Api();
+    this.setupEmbarkJS();
 
-    this.events.on("blockchain:started", this.registerWeb3Object.bind(this));
-    embark.registerActionForEvent("pipeline:generateAll:before", this.addWeb3Artifact.bind(this));
     embark.registerActionForEvent("deployment:contract:deployed", this.registerInVm.bind(this));
     embark.registerActionForEvent("deployment:contract:undeployed", this.registerInVm.bind(this));
     embark.registerActionForEvent("deployment:contract:deployed", this.registerArtifact.bind(this));
     embark.registerActionForEvent("deployment:contract:undeployed", this.registerArtifact.bind(this));
+  }
+
+  async setupEmbarkJS() {
+    this.events.request("embarkjs:plugin:register", 'blockchain', 'web3', 'embarkjs-web3');
+    this.events.request("embarkjs:console:register", 'blockchain', 'web3', 'embarkjs-web3');
```

Comment:
Shouldn't this be an async method with:
```suggestion
    await this.events.request2("embarkjs:console:register", 'blockchain', 'web3', 'embarkjs-web3');
```
?

Reply:
it doesn't need per se, but can be done

Reply:
To be prudent, `embarkjs:console:register` would need to be `await`ed before `embarkjs:console:setProvider` is called so that we know for sure the EmbarkJS provider registration has completed in the console before we call `setProvider`.